addons/mrp/models/mrp_production.py

```diff
@@ -529,7 +535,7 @@ def write(self, vals):
                 if production.workorder_ids and not self.env.context.get('force_date', False):
                     raise UserError(_('You cannot move a planned manufacturing order.'))
             if 'move_raw_ids' in vals and production.state != 'draft':
-                production.move_raw_ids.filtered(lambda m: m.state == 'draft')._action_confirm()
+                self._autoconfirm_production()
```

Comment:
either use production._autoconfirm_production() there or self._autoconfirm_production() out of the loop